core/interfaces/i_ast_node_location.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The interface for an AST node location.
+ * @author samelh@google.com (Sam El-Husseini)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IASTNodeLocation');
+
+/**
+ * An AST node location interface.
+ * @interface
+ */
+Blockly.IASTNodeLocation = function() {};
```

Comment:
Aside: I am skeptical about this empty interface.  Knowing that `X` implements `IASTNodeLocation` tells us precisely nothing about `X`.  Also, TypeScript is mostly structurally typed, so (as I understand it) _any_ (object) value could be assigned to a variable of type `IASTNodeLocation` without triggering a type error.

This leads me to wonder: is this type actually needed?  This PR would not be the place to remove it, but perhaps an issue should be filed to consider its removal.